app/views/menus/show.html.haml

```diff
@@ -4,12 +4,16 @@
 %ul
   = render @facade.orders
 
-%p= "#{t('views.menus.total_cost')}: #{@facade.total_cost}"
+%p= "#{t('views.menus.total_cost')}: #{@facade.total_cost}" if @facade.menu.show_price?(current_user)
 
-= render 'orders/form', order: @facade.new_order
+- if @facade.menu.placed_orer?(current_user)
```

Comment:
http://www.virtuouscode.com/2011/07/05/demeter-its-not-just-a-good-idea-its-the-law/